### ISSUE TITLE
Not able to setup environment variables before the debugger is launched

### ISSUE THREAD

**Issue by Ampaex** [microsoft/vscode-cpptools#10618]
### Environment

- OS and version: Kubuntu / linux 5.15.0-52-generic
- VS Code: 1.75.0
- C/C++ extension: 1.14.4
- OS and version of remote machine (if applicable):
- GDB / LLDB version: 

### Bug Summary and Steps to Reproduce

Bug Summary:
I'm trying to run a custom gdb(qnx neutrino -  ntoaarch64-gdb), but for that, I need to set up some environment variables before. I've tried to use the "environment" field, the "envFile" approach, using  "terminal.integrated.env.linux", doing a "preLaunchTask" and many other things but I'm not able to set them before the call of the binary. Every try results in gdb being executed without the environment variables.
I currently have an environment script(.sh) for setting up all the variables.

Steps to reproduce:
1. I'm trying to run the launch gdb that needs some environment variables to be set.
2. Try to add the envFile, environment field or any other approach.
3. Launch the debugger
4. See that the gdb debugger fails because variables are not set.

Could you help me with this issue?
Thanks

### Debugger Configurations

```shell
{
          "name": "Launch debug (GDB)",
          "type": "cppdbg",
          "request": "launch",
          "miDebuggerPath": "/path/to/ntoaarch64-gdb",
          "program": "/path/to/binary",
          "stopAtEntry": false,
          "cwd": "${workspaceFolder}",
          "miDebuggerServerAddress": "(ip):(port)",
          "externalConsole": true,
          "MIMode": "gdb",
          "setupCommands": [
            {
                "description": "Enable pretty-printing for gdb",
                "text": "-enable-pretty-printing",
                "ignoreFailures": true
            }
          ],
          "envFile": "${workspaceFolder}/.env",
          "environment": [
              {
                "name":"ENVVAR",
                "value": "/some/path/"
              }
           ]
        }
```

### Debugger Logs

```shell
¡QNX environment is not set!
```

### Other Extensions

I'm using the extensions for C/C++.

**Comment by pawelszramowski** [comment]
If the required environment variables do not depend on your VScode workspace then you could create an adapter script that would set the variables and call GDB, passing through to it all the command-line arguments. Untested example:

* `"miDebuggerPath": "/path/to/ntoaarch64-gdb-vscode"`
* `/path/to/ntoaarch64-gdb-vscode`:
  ```shell
  #!/usr/bin/env sh
  export NAME=VALUE
  exec ntoaarch64-gdb "$@"
  ```

**Comment by Ampaex** [comment]
Yes, could be an option, but in my case is not so useful because has to be run in multiple machines. I not a good idea to be modifying the installation on each of them

**Comment by pawelszramowski** [comment]
Right, the location of the script doesn't really matter, it may (and probably should) be versioned within the workspace. I'd then just make sure to change the current working directory to `/path/to/` before calling `ntoaarch64-gdb`, just like VSCode does.
